.github/workflows/repro.yml

```diff
@@ -0,0 +1,63 @@
+---
+# https://docs.corelightning.org/docs/repro
+name: Repro Build Nightly
+on:
+  # 05:00 Berlin, 03:00 UTC, 23:00 New York, 20:00 Los Angeles
+  schedule:
+    - cron: "0 3 * * *"
+jobs:
+  ubuntu:
+    name: "Ubuntu repro build: ${{ matrix.version }}"
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false    # Let each build finish.
+      matrix:
+        version: ['focal', 'jammy', 'noble']
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+
+      - name: Build environment setup
+        run: |
+          echo "Building base image for ${{ matrix.version }}"
+          sudo docker run --rm -v $(pwd):/build ubuntu:${{ matrix.version }} bash -c "\
+              apt-get update && \
+              apt-get install -y debootstrap && \
+              debootstrap ${{ matrix.version }} /build/${{ matrix.version }}"
+          sudo tar -C ${{ matrix.version }} -c . | docker import - ${{ matrix.version }}
+
+      - name: Builder image setup
+        run: docker build -t cl-repro-${{ matrix.version }} - < contrib/reprobuild/Dockerfile.${{ matrix.version }}
+
+      - name: Build using the builder image and store Git state
+        run: |
+          # Create release directory.
+          mkdir $GITHUB_WORKSPACE/release
+
+          # Perform the repro build.
+          docker run --name cl-build -v $GITHUB_WORKSPACE:/repo -e FORCE_MTIME=$(date +%F) -t cl-repro-${{ matrix.version }}
+
+          # Commit the image and use it to inspect the Git tree state.
+          docker commit cl-build cl-release
+          docker run --rm -v $GITHUB_WORKSPACE:/repo -t cl-release bash -c "\
+              git --no-pager status > /repo/release/git.log && \
+              git --no-pager diff >> /repo/release/git.log"
+
+          # Change permissions on the release files for access by the runner environment.
+          sudo chown -R runner $GITHUB_WORKSPACE/release
+
+      - name: Assert clean release
+        run: |
+          releasefile=$(ls release/clightning-*)
+          echo 'Release file:'
+          ls -al release/clightning-*
+
+          if [ -n "$(echo $releasefile | sed -n '/-modded/p')" ]; then
+            echo "Git Status and Diff:"
+            cat release/git.log
+
+            echo 'Error: release modded / dirty tree.'
+            exit 1
+          else
+            echo 'Success! Clean release.'
```

Comment:
The action is always passing for me ([workflow](https://github.com/ShahanaFarooqui/lightning/actions/workflows/repro.yml)). I removed commit 21fcd4594b9db99017e6c6f4cca750566d1b2969 from my [branch](https://github.com/ShahanaFarooqui/lightning/tree/repro-nightly-builds) for the [third run](https://github.com/ShahanaFarooqui/lightning/actions/runs/11041317409), which I expected to cause a dirty tree, but the job still succeeded.

Could you provide steps to replicate a failed run for testing?